lib.rs

```diff
@@ -310,6 +333,43 @@ impl ThreadPool {
             }
         }
     }
+    
+    /// Block the current thread until all jobs in the pool are completed.
+    /// Once waiting for the pool to complete you can no longer add new jobs, &mut self ensures that.
+    ///
+    /// ```
+    /// # use threadpool::ThreadPool;
```

Comment:
I think we should uncomment this so we have a complete example a user can just copy/paste